### ISSUE TITLE
Updated to 3.1.1 (2020-10-17) now Homebridge fails

### ISSUE THREAD

**Issue by robertfgoodworth** [mantorok1/homebridge-rinnai-touch-platform#2]
Grrr,  I just Updated to 3.1.1 (2020-10-17) now Homebridge fails with the following,  please help.

********
[10/19/2020, 8:39:25 PM] Homebridge v1.2.3 is running on port 52391.
[10/19/2020, 8:39:28 PM] Error [ERR_UNHANDLED_ERROR]: Unhandled error. ('connect EHOSTUNREACH 192.168.10.172:27847')
    at TcpService.emit (events.js:304:17)
    at Socket.<anonymous> (/homebridge/node_modules/homebridge-rinnai-touch-platform/src/rinnai/TcpService.ts:50:16)
    at Object.onceWrapper (events.js:422:26)
    at Socket.emit (events.js:315:20)
    at emitErrorNT (internal/streams/destroy.js:92:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
[10/19/2020, 8:39:28 PM] Got SIGTERM, shutting down Homebridge...
[10/19/2020, 8:39:28 PM] [Rinnai Touch] Shutting down plugin
********

Regards

Rob

**Comment by robertfgoodworth** [comment]
FYI,  I reverted back to version 3.1.0, all good again.  Sorry but there seems to be a BIG bug in version 3.1.1.

**Comment by mantorok1** [comment]
Hi Rob, sorry to hear the upgrade didn't work. 3.1.1 didn't contain any changes to the handling of the TCP connection to the Rinnai module, it was mainly fixing issues with Evap Cooling.

The module is prone to having connectivity issues. When this happens to me I find the following will fix it up:
1. Shutdown Homebridge and force quit the Rinnai Touch App on your phone (if necessary). You need to ensure nothing else attempts to connect to the module via WiFi
2. Restart your WiFi router or the Rinnai module (whichever is easier)
3. After a few minutes restart Homebridge

When you get a chance give 3.1.1 another go.

Regards

**Comment by mantorok1** [comment]
Hi again, good news. I have identified a bug in the way connection errors are handled. Fixing this will prevent the plugin from crashing when it fails to connect to the Rinnai module. Once fixed the plugin will attempt to connect 3 times and then wait 1 minute before trying again.  However, it will not fix the cause of the connection error. As mentioned above you'll need to restart your router or module to fix this.

**Comment by robertfgoodworth** [comment]
Hi Mantorok1,

 

Thanks for responding so quickly.

 

Yeah I suspected it is this flaky Rinnai Touch Wi-Fi, I’m going to lodge a service request with Rinnai to have it replaced, even before I used it in Homebridge (Home assistant) there touch app worked only sometimes.  I use Unifi Ubiquiti network hardware and this touch Wi-Fi shows latency issues even though it has good Wi-Fi connection (70%) on the same vlan, and it’s only 10m from one of the unifi Wi-Fi AP devices (3 throughout the house).

 

How close do you have to have the touch Wi-Fi from your Wi-Fi AP?  More so I’m interested to know what your ping responses are, mine range from 50 to 200 ms, up and down like a yoyo. Done of my other 50+ Wi-Fi devices show this trend and all have good ping responses (under 20ms).

 

Anyhow, It finally managed to connect again,  I’ve updated to 3.1.1,  then this touch Wi-Fi went mad again for 20mins then finally made one good connection before it started it’s network connection issues again.

 

It definitely an issue with the touch Wi-Fi and not your plugin.  Again thanks for you response.

 

Regards Robert

**Comment by mantorok1** [comment]
My module is installed in the roof cavity and is about 3 - 4m from the router. Ping responses seem to be similar to what you are seeing but I'm not getting random connection dropouts. Once the plugin connects it remains very stable.

```
64 bytes from 192.168.1.3: icmp_seq=0 ttl=128 time=239.679 ms
64 bytes from 192.168.1.3: icmp_seq=1 ttl=128 time=136.089 ms
64 bytes from 192.168.1.3: icmp_seq=2 ttl=128 time=147.743 ms
64 bytes from 192.168.1.3: icmp_seq=3 ttl=128 time=203.776 ms
64 bytes from 192.168.1.3: icmp_seq=4 ttl=128 time=62.756 ms
64 bytes from 192.168.1.3: icmp_seq=5 ttl=128 time=385.555 ms
64 bytes from 192.168.1.3: icmp_seq=6 ttl=128 time=97.033 ms
64 bytes from 192.168.1.3: icmp_seq=7 ttl=128 time=24.535 ms
64 bytes from 192.168.1.3: icmp_seq=8 ttl=128 time=39.829 ms
64 bytes from 192.168.1.3: icmp_seq=9 ttl=128 time=61.183 ms
64 bytes from 192.168.1.3: icmp_seq=10 ttl=128 time=78.867 ms
```

**Comment by mantorok1** [comment]
Hi @robertfgoodworth, I've released a new version of the plugin which should prevent the plugin from crashing when it fails to connect to the module. Let me know if this fixes the issue for you and I'll close it off.

Regards

**Comment by robertfgoodworth** [comment]
Hi Mantorok,

 

Awesome work,  did the trick, does not crash now, just loses connection all the time ☹.

 

 

 



 

 

Crazy hey ☹

 

I’ve lodged a service request with Brivis, even my NC7 touch shows error 53 quite often.

 

Cheers

 

Rob

**Comment by mantorok1** [comment]
The Rinnai module uses 11g (2.4GHz). Is it possible you are getting some interference such as a neighbour's WiFi or your microwave oven? Have you tried setting different channels on your router for the 2.4GHz band?

**Comment by robertfgoodworth** [comment]
Hi Mantorok1,

 

Thanks for the feedback, yes I’m very aware of Wifi channels etc, My Unifi Ubiquity Cloud Key 2 checks my wifi channel usage every Sunday night and if needed changes to the least used channels for both Ghz ranges,  I also have around 60 devices on Zigbee (2.4 channel 15). So far of all 108 devices on my networks (inc zigbee) this device is the only one with these network latency issues.

 

Perhaps a new Touch wifi module may help.  

 

Do you know if Rinnai have ever released a firmware update for these? 

 

Cheers

 

Rob

**Comment by mantorok1** [comment]
I'm not aware of any firmware updates for the module.

**Comment by mantorok1** [comment]
This issue has been fixed in v3.1.2

**Comment by robertfgoodworth** [comment]
Just a FYI and in case someone else has this issue.

 

I seem to have stablised it by moving the Touch Wfi box 2 meter from my Unifi Wifi AP, comms seems to be more stable. Talk about crappy wifi in there devices.

 

Cheers

 

Rob.
